packages/babel-traverse/src/scope/index.ts

```diff
@@ -1069,15 +1069,38 @@ export default class Scope {
       path = (this.getFunctionParent() || this.getProgramParent()).path;
     }
 
+    const { init, unique, kind = "var", id } = opts;
+
+    // When injecting a non-const non-initialized binding inside
+    // an IIFE, if the number of call arguments is less than or
+    // equal to the number of function parameters, we can safely
+    // inject the binding into the parameter list.
+    if (
+      !init &&
+      !unique &&
+      kind !== "const" &&
```

Comment:
Can we compare with `var` and `let` only? So it can handle the const-like using declarations.

Reply:
```suggestion
      (kind === "var" || kind === "let") &&
```